src/transformers/pipelines/base.py

```diff
@@ -50,6 +55,11 @@
 logger = logging.get_logger(__name__)
 
 
+def collate_fn(items):
+    assert len(items) == 1, "This collate_fn is meant to be used with batch_size=1"
```

Comment:
Nit: we are moving away from asserts in the code base, so please use and if xxx, raise ValueError.

Reply:
Got it ! Any reason why ?  (Just curious if there's a background on that decision)

Reply:
See internal slack. The main gist is: asserts are more meant for debugging and will be ignored if you use python in optimized mode. This is not a debugging statement, so it should not use an assert.